src/GovUk.Education.ExploreEducationStatistics.Content.Api.Tests/Controllers/RedirectsControllerTests.cs

```diff
@@ -0,0 +1,252 @@
+                new(FromSlug: "methodology_fromSlug_2", ToSlug: "methodology_toSlug_2"),
+                new(FromSlug: "methodology_fromSlug_3", ToSlug: "methodology_toSlug_3")
+            };
+
+            var cachedViewModel = new RedirectsViewModel(
+                Publications: cachedPublicationRedirects,
+                Releases: cachedReleaseRedirects,
+                Methodologies: cachedMethodologyRedirects);
+
+            await blobCacheService.SetItemAsync(new RedirectsCacheKey(), cachedViewModel);
+
+            var response = await ListRedirects(client);
+
+            var viewModel = response.AssertOk<RedirectsViewModel>();
+
+            Assert.All(
+                cachedPublicationRedirects,
+                cpr => Assert.Contains(
+                    viewModel.Publications,
+                    rvm => cpr.FromSlug == rvm.FromSlug && cpr.ToSlug == rvm.ToSlug));
+            Assert.All(
+                cachedReleaseRedirects,
+                crr => Assert.Contains(
+                    viewModel.Releases,
+                    rvm => crr.FromSlug == rvm.FromSlug && crr.ToSlug == rvm.ToSlug));
+            Assert.All(
+                cachedMethodologyRedirects,
+                cmr => Assert.Contains(
+                    viewModel.Methodologies,
+                    rvm => cmr.FromSlug == rvm.FromSlug && cmr.ToSlug == rvm.ToSlug));
+        }
+
+        [Fact]
+        public async Task NoRedirectsExist_Returns200WithNoRedirects()
+        {
+            var contentDbContext = ContentDbContextMock();
+
+            var client = BuildApp(contentDbContext.Object).CreateClient();
+
+            var response = await ListRedirects(client);
+
+            var viewModel = response.AssertOk<RedirectsViewModel>();
+
+            Assert.Empty(viewModel.Publications);
+            Assert.Empty(viewModel.Releases);
+            Assert.Empty(viewModel.Methodologies);
+        }
+
+        private static Mock<ContentDbContext> ContentDbContextMock(
+            IEnumerable<PublicationRedirect>? publicationRedirects = null,
+            IEnumerable<ReleaseRedirect>? releaseRedirects = null,
+            IEnumerable<MethodologyRedirect>? methodologiesRedirects = null)
+        {
+            var contentDbContext = new Mock<ContentDbContext>();
+
+            contentDbContext.Setup(context => context.PublicationRedirects)
+                .Returns((publicationRedirects ?? Array.Empty<PublicationRedirect>()).AsQueryable().BuildMockDbSet().Object);
+
+            contentDbContext.Setup(context => context.ReleaseRedirects)
+                .Returns((releaseRedirects ?? Array.Empty<ReleaseRedirect>()).AsQueryable().BuildMockDbSet().Object);
+
+            contentDbContext.Setup(context => context.MethodologyRedirects)
+                .Returns((methodologiesRedirects ?? Array.Empty<MethodologyRedirect>()).AsQueryable().BuildMockDbSet().Object);
+
+            return contentDbContext;
+        }
```

Comment:
Not sure that we really want to do this here, as most other integration tests are using the EF Core In-Memory database for `ContentDbContext`.

I think you probably want to be using `await TestApp.AddTestData<ContentDbContext>` instead.

This pattern looks like it could have been copied from `DataSetFilesControllerTests`. The reason that class uses `ContentDbContextMock` is because `ReleaseFilesFreeTextTable` used by it isn't supported by EF Core In-Memory.